model-optimizer/extensions/ops/interpolate_test.py

```diff
@@ -0,0 +1,275 @@
+                 [20, 40, 1380], [20.0, 40.0 / 85.0, 1380.0 / 690.0], [0, 2, 4]),
+                ([4, 3, 11, 22, 5], [1, 3, 4, 8, 5], [1, 16, 85, 470, 690], [60, 22, 430, 500, 345],
+                 [60, 430, 345], [10.0, 4.3, 345.0 / 700.0], [0, 2, 4]),
+                ([0], [0], [5, 77, 444, 88, 6050], [100, 308, 4440, 44, 6050],
+                 [100, 308, 4440, 44], [20.0, 4.0, 10.0, 0.5], [0, 1, 2, 3]),
+                ([0], [0], [1, 100, 200], [1, 350, 150], [350, 150], [3.5, 150 / 200], [1, 2]),
+                ([0, 3, 10], [0], [16, 7, 190], [8, 10, 390], [8, 390], [0.5, 390 / 200], [0, 2]),
+                ([10, 0, 10], [0, 16, 18], [4, 1024, 8000], [56, 520, 8028], [56, 520], [4.0, 0.5], [0, 1]),
+                ([0], [0], [1, 690], [20, 1380], [20, 1380], [20.0, 1380.0 / 690.0], [0, 1]),
+                ([4, 3, 11, 22, 5, 0], [1, 3, 4, 8, 5, 0], [1, 16, 85, 470, 690, 349], [60, 22, 430, 500, 345, 349],
+                 [60, 430, 345], [10.0, 4.3, 345.0 / 700.0], [0, 2, 4])
+                ])
+    def test_interpolate4_using_sizes(self, pads_begin, pads_end, input_shape, output_shape, sizes, scales, axes):
+        graph = build_graph(nodes_attrs=graph_nodes_attrs,
+                            edges=graph_edges,
+                            update_attributes={
+                                'input_data': {'shape': input_shape},
+                                'sizes': {'shape': int64_array(sizes).shape, 'value': int64_array(sizes)},
+                                'sizes_data': {'shape': int64_array(sizes).shape, 'value': int64_array(sizes)},
+                                'scales': {'shape': np.array(scales).shape, 'value': np.array(scales)},
+                                'scales_data': {'shape': np.array(scales).shape, 'value': np.array(scales)},
+                                'axes': {'shape': int64_array(axes).shape, 'value': int64_array(axes)},
+                                'axes_data': {'shape': int64_array(axes).shape, 'value': int64_array(axes)},
+                                'interpolate': {'pads_begin': int64_array(pads_begin),
+                                                'pads_end': int64_array(pads_end)}
+                            })
+
+        node = Node(graph, 'interpolate')
+        tested_class = Interpolate(graph=graph, attrs=node.attrs())
+        tested_class.infer(node)
+
+        msg = "Interpolate-4 infer failed for case: sizes={}, scales={}, pads_begin={}, pads_end={}, axes={}," \
+              " expected_shape={}, actual_shape={}"
+
+        self.assertTrue(np.array_equal(graph.node['interpolate_data']['shape'], int64_array(output_shape)),
+                        msg.format(sizes, scales, pads_begin, pads_end, axes, output_shape,
+                                   graph.node['interpolate_data']['shape']))
+
+    @generate(*[([0], [0], [1, 3, 100, 200], [1, 3, 350, 150], [350, 150], [3.5, 150 / 200], [2, 3]),
```

Comment:
Add a case where `scales = 0.3333333` for one of the dimensions

Reply:
Done.